### ISSUE TITLE
Expose `AbstractScheduledEventExecutor.getCurrentTimeNanos`

### ISSUE THREAD

**Issue by yawkat** [netty/netty#12827]
It would be useful to expose the `getCurrentTimeNanos` method to channel handlers, so that they can use it instead of calling `System.nanoTime` directly. The main purpose would be to make handlers work predictably with the EmbeddedEventLoop time manipulation methods (advance, freeze).

A good example is IdleStateHandler. It uses both event loop scheduling, and then nanoTime to confirm the timeout. If the event loop was advanced with `advanceBy`, the scheduled task will run, but the nanoTime will not show a timeout. For the unit test of IdleStateHandler, this nanoTime call is already overridden (it's in a package-private method), but when writing an external test that involves IdleStateHandler, this is not available. By exposing `getCurrentTimeNanos`, we would have a "single source of truth" when it comes to time, circumventing this issue.

Another minor advantage would be that handlers can take advantage of the nanoTime offset that the event loops implement, but this is not worth a change on its own.

The right place for this api would probably be `EventExecutor` (maybe `EventExecutorGroup`? does it make sense to have different times for each event loop in a group?). What do you think, is this worth adding?

**Comment by yawkat** [comment]
Project reactor has a similar idea in its Scheduler class: https://projectreactor.io/docs/core/release/api/reactor/core/scheduler/Scheduler.html#now-java.util.concurrent.TimeUnit-

**Comment by normanmaurer** [comment]
I need to think a bit more about it

**Comment by trustin** [comment]
How about introducing something similiar to `Ticker` in Guava? - https://guava.dev/releases/21.0/api/docs/com/google/common/base/Ticker.html Much simpler than Reactor's `Scheduler` yet does its job. We may want to add a few more methods though, e.g. `advance()` and `sleep()`.

**Comment by trustin** [comment]
It's actually something I wanted to implement in our codebase to reduce the overall test run time. Let me try to come up with one soon.

**Comment by yawkat** [comment]
@trustin i think exposing getCurrentTimeNanos would be basically the same as what you describe. Since #12459, advance already exists for the embedded event loop. So I'm not sure what you're proposing. Do you want to make it independent of EmbeddedEventLoop?

**Comment by trustin** [comment]
> i think exposing getCurrentTimeNanos would be basically the same as what you describe. Since https://github.com/netty/netty/pull/12459, advance already exists for the embedded event loop. So I'm not sure what you're proposing. Do you want to make it independent of EmbeddedEventLoop?

Correct. I want to make it generic enough so it can be used in more places than just `EmbeddedEventLoop`.

**Comment by trustin** [comment]
@yawkat I've just sent out a PR for my idea. Would you mind taking a look? #13169
